src/utils/automation-helper.js

```diff
@@ -81,90 +83,93 @@ export const parseOutput = (data, translator) => {
 };
 
 // eslint-disable-next-line consistent-return, no-async-promise-executor
-export const automationRunPolling = async (automationId, runId, translator, time = 0) => {
+export const automationRunPolling = async (automationId, runId, translator) => {
   const runningStatuses = ['queued', 'running', 'not started', 'starting'];
-  const pollInterval = 100;
   const defaultMessage = getTranslation(
     translator,
     'Object.ActionButton.Automation.DefaultAutomationMsg',
     'Automation finished'
   );
-  const automationRun = await getAutomationRun(automationId, runId);
-  const { status } = automationRun;
-  if (runningStatuses.includes(status)) {
-    if (time > 10 * 60 * 1000) {
-      return { ok: false, message: getTranslation(translator, 'geo.findLocation.error.timeout', 'Timeout') };
+  let status = 'queued';
+  let automationRun;
+  const endTime = Date.now() + MAX_POLL_TIME;
+  while (runningStatuses.includes(status)) {
+      if (Date.now() > endTime) {
+        return { ok: false, message: getTranslation(translator, 'geo.findLocation.error.timeout', 'Timeout') };
+      }
+      // eslint-disable-next-line no-await-in-loop
+      automationRun = await getAutomationRun(automationId, runId);
+      status = automationRun.status;
+      // eslint-disable-next-line no-await-in-loop, no-promise-executor-return
+      await new Promise((resolve) => setTimeout(resolve, POLL_INTERVAL));
     }
-    setTimeout(automationRunPolling(automationId, runId, translator, time + pollInterval), pollInterval);
-  } else {
-    let msg;
-    switch (automationRun.status) {
-      case 'finished': {
-        if (automationRun.title?.length > 0) {
-          msg = parseOutput(automationRun.title, translator);
-          msg.ok = true;
-        } else {
-          msg = {
-            message: defaultMessage,
-            ok: true,
-          };
-        }
-        break;
+  let msg;
+  switch (automationRun.status) {
```

Comment:
In general it's going to be easier to follow the code path if you just return when the output is available. Instead of re-assigning a variable. This also reducs the code complexity by removing the need for the `else` statement.


Example
```js
switch (automationRun.status) {
    case 'finished': {
      if (automationRun.title?.length > 0) {
       return { ...parseOutput(automationRun.title, translator), ok: true };
      }

     return {
          message: defaultMessage,
          ok: true,
        };
    }
```

Consider this an optional improvement.